xsnap/documentation/StaticModuleRecord.md

```diff
@@ -0,0 +1,181 @@
+# Static Module Records
+
+Revised: January 6, 2022
+
+Warning: These notes are preliminary. Omissions and errors are likely. If you encounter problems, please ask for assistance.
+
+## About
+
+A static module record prepares a module descriptor to be loaded and initialized as a new module into a compartment.
+
+> Compartments can use module descriptors directly instead of indirectly thru static module records.
+
+A static module record also provides access to the module bindings.
+
+## The StaticModuleRecord Constructor
+
+### StaticModuleRecord(descriptor)
+
+Returns a static module record object. Currently XS supports two kinds of module descriptor.
+
+### descriptor.source
+
+The `source` module descriptor constructs a static module record from a string that is parsed and compiled as a module. 
+
+	const smr = new StaticModuleRecord({ source: `
+		import x from "mod";
+		export let y = x;
+	` });
```

Comment:
Would it be possible for this form to accept a string alone as an overload? That would eliminate a gap in implementation and proposal.